packages/app-store/btcpayserver/pages/setup/index.tsx

```diff
@@ -0,0 +1,349 @@
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import { useCompatSearchParams } from "@calcom/lib/hooks/useCompatSearchParams";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { trpc } from "@calcom/trpc";
+import { Button } from "@calcom/ui/components/button";
+import { Icon } from "@calcom/ui/components/icon";
+import { showToast } from "@calcom/ui/components/toast";
+
+import KeyField from "../../components/KeyInput";
+import { btcpayCredentialKeysSchema } from "../../lib/btcpayCredentialKeysSchema";
+
+export type IBTCPaySetupProps = z.infer<typeof btcpayCredentialKeysSchema>;
+
+export default function BTCPaySetup(props: IBTCPaySetupProps) {
+  const params = useCompatSearchParams();
+  if (params?.get("callback") === "true") {
+    return <BTCPaySetupCallback />;
+  }
+  return <BTCPaySetupPage {...props} />;
+}
+
+function BTCPaySetupCallback() {
+  const [error, setError] = useState<string | null>(null);
+  const searchParams = useCompatSearchParams();
+
+  useEffect(() => {
+    if (!searchParams) {
+      return;
+    }
+    if (!window.opener) {
+      setError("Something went wrong. Opener not available. Please contact support");
+      return;
+    }
+    const code = searchParams?.get("code");
+    const error = searchParams?.get("error");
+
+    if (!code) {
+      setError("declined");
+    }
+    if (error) {
+      setError(error);
+      return;
+    }
+
+    window.opener.postMessage({
+      type: "btcpayserver:oauth:success",
+      payload: { code },
+    });
+    window.close();
+  }, [searchParams]);
+
+  return (
+    <div>
+      {error && <p>Authorization failed: {error}</p>}
+      {!error && <p>Connecting...</p>}
+    </div>
+  );
+}
+
+function BTCPaySetupPage(props: IBTCPaySetupProps) {
+  const router = useRouter();
+  const { t } = useLocale();
+  const session = useSession();
+  const [loading, setLoading] = useState<boolean>(false);
+  const [validating, setValidating] = useState<boolean>(false);
+  const [updatable, setUpdatable] = useState<boolean>(false);
+  const [keyData, setKeyData] = useState<
+    | {
+        storeId: string;
+        serverUrl: string;
+        apiKey: string;
+        webhookSecret: string;
+      }
+    | undefined
+  >();
+  const settingsSchema = z.object({
+    storeId: z.string().trim(),
+    serverUrl: z.string().trim(),
+    apiKey: z.string().trim(),
+    webhookSecret: z.string().optional(),
+  });
+  const integrations = trpc.viewer.apps.integrations.useQuery({ variant: "payment", appId: "btcpayserver" });
+  const [btcPayPaymentAppCredentials] = integrations.data?.items || [];
+  const [credentialId] = btcPayPaymentAppCredentials?.userCredentialIds || [-1];
+  const showContent = !!integrations.data && integrations.isSuccess && !!credentialId;
+
+  const saveKeysMutation = trpc.viewer.apps.updateAppCredentials.useMutation({
+    onSuccess: () => {
+      showToast(t("keys_have_been_saved"), "success");
+      router.push("/event-types");

```

Comment:
Instead of redirecting to event-types. We should redirect to the same event's App tab, so that user can continue configuring the recently installed app.
<img width="2716" height="1062" alt="image" src="https://github.com/user-attachments/assets/55dc4e22-3c8a-4076-973d-d668a21fd98a" />

Reply:
For this, once the keys are saved, that's all for the config. Nothing more is required unless setting events to receive payment.

Reply:
Yeah that is the point, without user enabling this App for the event and setting up the price, they won't be able to use the app.

 So, when after the setup, user is thrown back to /event-types, user doesn't know what to do now. So, it is best to redirect him back to this page(Apps tab) so that he can enable the App.